biz.aQute.bndlib/src/aQute/bnd/osgi/resource/ResourceUtils.java

```diff
@@ -546,6 +546,8 @@ public static VersionedClause toVersionClause(Resource resource, String mask) {
 		if (resource.getCapabilities(WORKSPACE_NAMESPACE).isEmpty()) {
 			Macro macro = new Macro(new Processor());
 			Version version = getVersion(idCap);
+			if (version == null)
+				version = Version.emptyVersion;
```

Comment:
I am not sure this is the right answer. I think we should test for null and throw an exception indicating malformed osgi.identity capability. Just replacing with 0 seems wrong.

Reply:
Also, I will note that `XMLResourceGenerator` is partly to blame for this mess. It simply writes out what it has in the memory Resource objects without regard to the validity of the document to the schema. So if the osgi.identity Resource object does not have a version attribute, `XMLResourceGenerator` does not write out `version:Version=0` as required by the schema.